### PR TITLE
python3Packages.certbot-dns-google: init at 1.20.0

### PR DESCRIPTION
**Pull request by jpetrucciani**
###### Motivation for this change

I would like to be able to use the GCP DNS plugin for Certbot to provision new LetsEncrypt certificates!

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/certbot-dns-google/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  checkInputs = [
+    pytest
+    pytestCheckHook
+  ];
+
+  disabled = !isPy3k;
+
+  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
+
+  sourceRoot = "source/${pname}";
```
[the remark this answers is not in the document]
Comment (jpetrucciani) [reply pkgs/development/python-modules/certbot-dns-google/default.nix]: I had just followed what other plugins appeared to do. I've swapped it for all plugins now.

File: pkgs/development/python-modules/certbot-dns-google/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+buildPythonPackage rec {
+  inherit (certbot) src version;
+
+  pname = "certbot-dns-google";
+
+  propagatedBuildInputs = [
+    acme
+    oauth2client
+    google-api-python-client
+    certbot
```
[the remark this answers is not in the document]
Comment (jpetrucciani) [reply pkgs/development/python-modules/certbot-dns-google/default.nix:19]: I believe this is needed as the library exists within the codebase of certbot - this is also done within the other plugins

**Comment by jpetrucciani** [comment]
@SuperSandro2000 addressed most of your comments, and applied some of that to the other certbot plugins in nixpkgs to keep it consistent. Also rebased off of master - however, now it seems to be broken on Darwin, at least when I try locally. I'm currently building it on a linux amd64 env

**Comment by jpetrucciani** [comment]
Works fine on Linux amd64! Do we normally wait for a fix upstream to something like BS4 before merging something like this?

**Review by jonringer — COMMENTED** [review]
I would separate these changes into two commits, as there's really 2 "logical" changes being done here

To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
python3Packages.certbot-dns-google: init at 1.20.0
python3Packages.certbot-dns-*: lint
```

**Comment by jpetrucciani** [comment]
@jonringer Thanks for the review! I can separate the commits here (also I think both links in your comment may be broken - they 404 for me at the moment)

**Comment by jonringer** [comment]
> @jonringer Thanks for the review! I can separate the commits here (also I think both links in your comment may be broken - they 404 for me at the moment)

oh yea, it was moved. thanks

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 142574 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>9 packages built:</summary>  <ul>    <li>certbot-full</li>    <li>python38Packages.certbot-dns-cloudflare</li>    <li>python38Packages.certbot-dns-google</li>    <li>python38Packages.certbot-dns-rfc2136</li>    <li>python38Packages.certbot-dns-route53</li>    <li>python39Packages.certbot-dns-cloudflare</li>    <li>python39Packages.certbot-dns-google</li>    <li>python39Packages.certbot-dns-rfc2136</li>    <li>python39Packages.certbot-dns-route53</li>  </ul></details>

**Pull request merged**